autonomi/src/client/high_level/files/fs_shared.rs

```diff
@@ -1,57 +1,200 @@
+// Copyright 2025 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+use super::CombinedChunks;
+use crate::client::high_level::data::DataAddress;
+use crate::client::payment::PaymentOption;
 use crate::client::payment::Receipt;
-use crate::client::{ClientEvent, UploadSummary};
+use crate::client::{ClientEvent, PutError, UploadSummary};
 use crate::files::UploadError;
 use crate::Client;
 use ant_evm::{Amount, AttoTokens};
+use ant_protocol::storage::{Chunk, DataTypes};
+use std::sync::LazyLock;
+
+type AggregatedChunks = Vec<((String, Option<DataAddress>, usize, usize), Chunk)>;
+
+/// Number of batch size of an entire quote-pay-upload flow to process.
+/// Suggested to be multiples of 80 (records-payouts-per-transaction).
+///
+/// Can be overridden by the `UPLOAD_FLOW_BATCH_SIZE` environment variable.
+static UPLOAD_FLOW_BATCH_SIZE: LazyLock<usize> = LazyLock::new(|| {
+    let batch_size = std::env::var("UPLOAD_FLOW_BATCH_SIZE")
+        .ok()
+        .and_then(|s| s.parse().ok())
+        .unwrap_or(80);
```

Comment:
`UPLOAD_FLOW_BATCH_SIZE` should default to `85`, or even better `MAX_TRANSFERS_PER_TRANSACTION / 3` (= 85 atm).